util-print.c

```diff
@@ -422,6 +422,60 @@ signed_relts_print(netdissect_options *ndo,
 	unsigned_relts_print(ndo, secs);
 }
 
+/*
+ * Format a struct tm with strftime().
+ * If the pointer to the struct tm is null, that means that the
+ * routine to convert a time_t to a struct tm failed; the localtime()
+ * and gmtime() in the Microsoft Visual Studio C library will fail,
+ * returning null, if the value is before the UNIX Epoch.
+ */
+const char *
+nd_format_time(char *buf, size_t bufsize, const char *format,
+         const struct tm *timeptr)
+{
+	if (timeptr != NULL) {
+		if (strftime(buf, bufsize, format, timeptr) != 0)
+			return (buf);
+		else
+			return ("[Does not fit in buffer]");
+	} else
+		return ("[Unrepresentable]");
```

Comment:
I was going to make a point that it might be more consistent to mimic `printf("%s", NULL)`, which prints a `(null)` string in similar cases, but apparently today on Linux is just segfaults like in old times, so the argument collapsed.

Also, if I was writing this code, I would flip `if` branches around to have the shortest hand first, and then would lose `else` after `return`, which would make the code easier to follow. But for such a small function it is a non-issue.